xbmc/cores/VideoPlayer/DVDSubtitles/SubtitlesStyle.h

```diff
@@ -115,15 +115,21 @@ struct renderOpts
 {
   float frameWidth;
   float frameHeight;
+  // Video size width, may be influenced by video settings (e.g. zoom)
   float videoWidth;
+  // Video size height, may be influenced by video settings (e.g. zoom)
   float videoHeight;
+  float videoScaledWidth; // Video size width, scaled to frame
```

Comment:
Do we really need all these new members? Afaics you compute a ratio based on the already existing members just to compute the scaled sizes based on the found ratio. If you could somehow do this on the flight without storing the values it would greatly simplify the code. Also we seem to passthrough always the same set of values (video rect, source rect, etc) everywhere

Reply:
yes i can move it within libass RenderImage to avoid add these members
i had done it this way to avoid the calculation at each frame

Reply:
i was thinking now, this is used to calculate sar/dar
but now with videoScaled* vars will results that have same ratio of source size
therefore now sar/dar will have always the same value
and `dar / sar` in `ass_set_pixel_aspect(m_renderer, dar / sar);` if i am not wrong should now results always as 1.0,
then we could set `ass_set_pixel_aspect(m_renderer, 1.0);` fixed value or else change as:
```
double sar = static_cast<double>(opts.sourceWidth) / static_cast<double>(opts.sourceHeight);
double dar = static_cast<double>(opts.sourceWidth) / static_cast<double>(opts.sourceHeight);
```

Reply:
Calibration window applies to GUI as a whole, in the past subtitles were rendered by OverlayRendererGUI which was effectively a GUI component. However setups with old 4:3 CRTs are still common nowadays. I wonder if this PR breaks them again (Ronie provided a way to simulate that in estuary a while ago but I don't remember the instructions by heart).
Maybe we could have another option in the calibration window to account for this.

Personally I feel the subtitle line in the calibration screen a bit confusing for someone new to kodi - you affect positioning while the effect on other properties like scale are a bit obscure. A rectangle positioned at the center of the line which you could move up/down to affect position and right/left to affect scale would capture all this much better and allow to force the overlays to be rendered inside the rendered rect if you want so...although at the cost of quite a bit of work. 

Displaying actual placeholder text would even be better (but even harder to implement).

Reply:
par is now implemented by taking value from "calibration window", remains to do some more tests

> Maybe we could have another option

would you like to provide a setting to set a customised aspect ratio for subtitles only separately? (maybe better xml advanced setting)
or did you think of other type of settings?

Reply:
Let's wait for user feedback and come back to this later :)